### ISSUE TITLE
errors with data overwrite

### ISSUE THREAD

**Issue by pqHelloWorld** [bufbuild/buf#3052]
When I was using "buf breaking", an error message appeared: "Failure: proto: not found; proto: not found; proto: not found; proto: not found; proto: not found; proto: not found".After reviewing the code, I noticed instances where the return value of "append" was being directly used as a parameter for recursive functions and in a loop where a slice was continuously being appended. This practice could potentially lead to unexpected issues, as the return value of "append" might not necessarily be a brand-new address, depending on whether the "append" operation has exceeded the capacity of the slice.like this:
private/bufpkg/bufprotosource/file.go
```
for nestedMessageIndex, nestedMessageDescriptorProto := range descriptorProto.GetNestedType() {
		nestedMessage, err := f.populateMessage(
			nestedMessageDescriptorProto,
			topLevelMessageIndex,
			append(nestedMessageIndexes, nestedMessageIndex),
			append(nestedMessageNames, message.Name()),
			message,
		)
		if err != nil {
			return nil, err
		}
		message.addNestedMessage(nestedMessage)
}
```

You can use the following proto as an experiment and when printing the "FullName" field, it will appear as "level1.level2.level3.level4.level5.level6.msg2.filed1".
```
message level1 {
  message level2 {
    message level3 {
      message level4 {
        message level5 {
          message level6 {
            message msg1 {
              int32 filed1 = 1;
            }
            message msg2 {
              int32 filed2 = 2;
            }
          }
        }
      }
    }
  }
}
```

**Comment by bufdev** [comment]
The error you are seeing is likely correct, and due to parallel execution. The full name you have spelled out is correct here. If you think there is a bug, we would appreciate a specific, reproducible (for example, a temporary GitHub repository containing the setup, along with a set of commands to reproduce the issue) and we will be happy to investigate!

**Comment by bufdev** [comment]
Separately - @doriable @oliversun9  we do have some of these append issues around the codebase - we should be making explicit copies of the slices before the recursive call, that was my fault. But that's unrelated to the issue here.

**Comment by bufdev** [comment]
This is fixed in v1.33.0.
